src/utils.ts

```diff
@@ -99,3 +99,15 @@ export function debug(text) {
     console.log(text);
   }
 }
+
+export function hashCode(str: string) {
```

Comment:
>hashCode [](start = 16, length = 8)

Very minor. Pretty weak hash function. Maybe find a better one?

Reply:
This is supposedly a hash function that is used in Java
http://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/